modules/sage/1.0/sage.smk

```diff
@@ -0,0 +1,255 @@
+        -ref_genome {input.fasta}
+        -hotspots {input.hotspots}
+        -panel_bed {input.panel_bed}
+        -high_confidence_bed {input.high_conf_bed}
+        -out {output.vcf}
+        >>  {log.stdout} 2> {log.stderr}
+        && bgzip -c {output.vcf} > {output.vcf_gz}
+        """)
+
+# Filter resulting VCF file on PASS variants
+rule _sage_filter_vcf:
+    input:
+        vcf = str(rules._run_sage.output.vcf_gz)
+    output:
+        vcf_passed = CFG["dirs"]["vcf"] + "combined/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/passed_combined.vcf.gz",
+        vcf_passed_tbi = CFG["dirs"]["vcf"] + "combined/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/passed_combined.vcf.gz.tbi"
+    log:
+        stdout = CFG["logs"]["vcf"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/filter_passed.stdout.log",
+        stderr = CFG["logs"]["vcf"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/filter_passed.stderr.log"
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    threads:
+        CFG["threads"]["filter"]
+    resources:
+        **CFG["resources"]["filter"]
+    shell:
+        op.as_one_line("""
+        bcftools view -f ".,PASS" {input.vcf} -Ov
+          | 
+        bcftools sort --max-mem {resources.mem_mb}M -Oz -o {output.vcf_passed}
+        > {log.stdout} 2> {log.stderr}
+          &&
+        tabix -p vcf {output.vcf_passed} >> {log.stdout} 2>> {log.stderr}
+        """)
+
+# Split filtered VCF file into snvs and indels
+rule _sage_split_vcf:
+    input:
+        vcf_passed = str(rules._sage_filter_vcf.output.vcf_passed)
+    output:
+        indels = CFG["dirs"]["vcf"] + "indels/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/indels.vcf.gz",
+        indels_tbi = CFG["dirs"]["vcf"] + "indels/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/indels.vcf.gz.tbi",
+        snvs = CFG["dirs"]["vcf"] + "snvs/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/snvs.vcf.gz",
+        snvs_tbi = CFG["dirs"]["vcf"] + "snvs/{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/snvs.vcf.gz.tbi",        
+    log:
+        stdout = CFG["logs"]["vcf"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/split_passed.stdout.log",
+        stderr = CFG["logs"]["vcf"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/split_passed.stderr.log"
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    threads:
+        CFG["threads"]["filter"]
+    resources:
+        **CFG["resources"]["filter"]
+    shell:
+        op.as_one_line("""
+        bcftools view -v indels {input.vcf_passed} -Oz -o {output.indels} > {log.stdout} 2> {log.stderr}
+          &&
+        tabix -p vcf {output.indels} >> {log.stdout} 2>> {log.stderr}
+          &&
+        bcftools view -v snps {input.vcf_passed} -Oz -o {output.snvs} >> {log.stdout} 2>> {log.stderr}
```

Comment:
Does the sum of the variants in the two outputs add up to the combined or do we lose some here?

Reply:
It does, but not always since I do not subset vcf file for mnvs. In other words, if there are no mnvs, the sum adds up to the total number of variants. If there are mnvs, then sum of the 2 outputs+number of mnvs adds up to the number in `combined` vcf.

Reply:
I see. So we lose mnvs if we use the individual split files. That's not ideal but I expect these are pretty rare, right?